core/src/main/java/org/apache/druid/indexer/partitions/SingleDimensionPartitionsSpec.java

```diff
@@ -170,6 +172,12 @@ public boolean isAssumeGrouped()
     return partitionDimension == null ? Collections.emptyList() : Collections.singletonList(partitionDimension);
   }
 
+  @Override
+  public String getForceGuaranteedRollupIncompatiblityReason()
+  {
+    return NAME + " partitions unsupported";
+  }
```

Comment:
should `isForceGuaranteedRollupCompatibleType` also be implemented to return false

Reply:
Perhaps it could be implemented in a clearer way, but currently I have the default implementation of `isForceGuaranteedRollupCompatibleType` in the `PartitionsSpec` interface that returns `true` if this method returns an empty string.